### ISSUE TITLE
Support dark mode

### ISSUE THREAD

**Issue by bubelov** [osmdroid/osmdroid#1804]
Not sure if it's even technically possible but it would be nice to have a built-in dark mode support

**Comment by spyhunter99** [comment]
There is a way to invert the tile colors, probably the best we can do.

**Comment by bubelov** [comment]
Might be related:

https://github.com/openstreetmap/openstreetmap-website/issues/2332

Looks like OSM website maintainers can't agree on how to implement it and they are also blocked by Bootstrap (for non-map components styling)

**Comment by henrichg** [comment]
I'm using this in Android application, when system is in dark mode:

```
MapView mMap ...;
mMap.setTileSource(TileSourceFactory.MAPNIK);
mMap.getOverlayManager().getTilesOverlay().setColorFilter(TilesOverlay.INVERT_COLORS);

```

**Comment by bubelov** [comment]
@henrichg I ended up doing the same thing, based on the sample app:

```kotlin
val nightMode =
    resources.configuration.uiMode and android.content.res.Configuration.UI_MODE_NIGHT_MASK

if (nightMode == android.content.res.Configuration.UI_MODE_NIGHT_YES) {
    overlayManager.tilesOverlay.apply {
        setColorFilter(TilesOverlay.INVERT_COLORS)
        loadingBackgroundColor = android.R.color.black
        loadingLineColor = Color.argb(255, 0, 255, 0)
    }
}
```

It works pretty well and it seem to be the recommended pattern so I'm wondering maybe it makes sense for osmdroid to do it automatically when it detects dark mode? It would be consistent with other widgets which tend to work out of the box.

The ocean looks a bit odd though:

![dark](https://user-images.githubusercontent.com/1189245/168831219-d1c80f83-66b1-4d3d-b887-98be0ceca1c3.jpg)

And here is how Google Maps shows it:

![darkgoog](https://user-images.githubusercontent.com/1189245/168831654-f99db3a6-c8df-4e6f-af1a-dd8e425b25c6.jpg)

**Comment by henrichg** [comment]
@bubelov yes, this also my own solution.

**Comment by spyhunter99** [comment]
google solves this problem, along with other tools like mapsforge, mapbox, etc by changing the way the map tiles are rendered. osmdroid only displays prerendered map tiles, we don't render the tiles on device. Beyond inverting the colors, the only other thing i can suggest is try finding a different tile source that has a night mode. Nothing else we can do, unless someone wants to add support for geotile json format and render on device.
